integration-tests/security/src/main/resources/application.properties

```diff
@@ -1,19 +1,21 @@
 # Authentication properties
 org.acme.openapi.security.auth.basic_auth/username=john
 org.acme.openapi.security.auth.basic_auth/password=test
+customAuth.additionalApiTypeAnnotations=@org.eclipse.microprofile.rest.client.annotation.RegisterClientHeaders,@org.eclipse.microprofile.rest.client.annotation.RegisterProvider(io.quarkiverse.openapi.generator.it.security.auth.DummyApiKeyAuthenticationProvider.class)
+quarkus.openapi-generator.codegen.enable-security-generation=false
 
 # since the file name has a space, we use the URI representation of this space here to not break the properties file
 # see the RFC3986 for more info https://datatracker.ietf.org/doc/html/rfc3986
 quarkus.openapi-generator.codegen.spec.open_weather_yaml.base-package=org.acme.openapi.weather.v2
+quarkus.openapi-generator.codegen.spec.open_weather_yaml.enable-security-generation=true
 quarkus.openapi-generator.open_weather_yaml.auth.app_id.api-key=12345
 
-quarkus.openapi-generator.codegen.spec.open_weather_v3_yaml.base-package=org.acme.openapi.weather.v3
-quarkus.openapi-generator.codegen.spec.open_weather_v3_yaml.client-headers-factory=none
-
-quarkus.openapi-generator.codegen.spec.open_weather_v4_yaml.base-package=org.acme.openapi.weather.v4
-quarkus.openapi-generator.codegen.spec.open_weather_v4_yaml.client-headers-factory=org.eclipse.microprofile.rest.client.ext.DefaultClientHeadersFactoryImpl
+quarkus.openapi-generator.codegen.spec.open_weather_custom_security_yaml.base-package=org.acme.openapi.weather.customsecurity
+quarkus.openapi-generator.codegen.spec.open_weather_custom_security_yaml.additional-api-type-annotations=${customAuth.additionalApiTypeAnnotations}
+quarkus.openapi-generator.open_weather_custom_security_yaml.auth.app_id.api-key=dummyKey
 
 quarkus.openapi-generator.codegen.spec.fooopenapi_json.base-package=org.acme.openapi.foo
+quarkus.openapi-generator.codegen.spec.fooopenapi_json.enable-security-generation=true
```

Comment:
This one we can leave `true`, so we are covered by the 3 scenarios.

Reply:
default value is used by open_weather_custom_security_yaml spec